### PR TITLE
test: improve visibility of console, network errors in cypress tests

### PR DESCRIPTION
**Pull request by mrrajamanickam-coveo**
* Replace stubs on console logs with spies so that messages are still logged to console to help debug tests
* Add log based on response status code of intercepted requests
* Add Cypress config to auto open browser dev console in headed mode
* Analytics intercept
  - Removed duplicate intercept, unified to analytics tracker
  - Added response status code check
  - Expanded URL regex to cover all downstream path segments

### TODO

- [x] Fix errors due to nested cy promise 
- [x] [Fix CI "Error: The Test Runner unexpectedly exited via a exit event with signal SIGSEGV"](https://coveord.atlassian.net/browse/KIT-2469)
- [ ] Fix [Search box test failures](https://github.com/coveo/ui-kit/pull/2872#issuecomment-1684471755) and any other valid TFs

https://coveord.atlassian.net/browse/KIT-2450

### CODE REVIEW

File: packages/atomic/cypress/fixtures/fixture-common.ts [hunk 1]
```diff
@@ -43,44 +43,46 @@ export const UrlParts = {
...
+export function setupIntercepts() {
+  function setupIntercept(path: string, alias: string, method = 'POST') {
+    cy.intercept(
+      {
+        method: method,
+        path: path,
+      },
+      (req) => {
+        req.continue((res) => {
+          // Not using expect() to cause tests to fail to allow for tests that exercise error cases.
```
Comment (mrrajamanickam-coveo) [inline packages/atomic/cypress/fixtures/fixture-common.ts:64]: But afaik tests that exercise error cases do their own intercepts and don't use the common intercepts - but just trying to be safe/conservative and not have this change destabilize tests any further

File: packages/atomic/cypress/fixtures/fixture-common.ts [hunk 2]
```diff
@@ -43,44 +43,51 @@ export const UrlParts = {
...
+            console.error(
+              `Request error on ${req.url} : ${res.statusCode} ${res.statusMessage}`
+            );
+          }
+        });
+      }
+    ).as(alias.substring(1));
+  }
+
+  setupIntercept('**/rest/v15/analytics/*', RouteAlias.UA, 'POST', false);
```
Comment (mrrajamanickam-coveo) [inline packages/atomic/cypress/fixtures/fixture-common.ts:80]: Notice that a number of reqs to analytics return 400 (even when tests pass) - is this expected ?

Comment (btaillon) [reply packages/atomic/cypress/fixtures/fixture-common.ts:80]: They shouldn't return 400 AFAIK 🤔 must be an assertion we forgot

**Review by btaillon — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Review by lvu285 — APPROVED** [review]

File: packages/atomic/cypress/fixtures/fixture-common.ts [hunk 3]
```diff
@@ -36,51 +36,58 @@ export const ConsoleAliases = {
...
+    cy.intercept(
+      {
+        method: method,
+        path: path,
+      },
+      (request) => {
+        request.continue((response) => checkResponseStatus(request, response));
+      }
+    ).as(alias.substring(1));
+  }
```
[the remark this answers is not in the document]
Comment (mrrajamanickam-coveo) [reply packages/atomic/cypress/fixtures/fixture-common.ts:80]: No particular reason other to contain scope as it is not currently used outside.

**Comment by mrrajamanickam-coveo** [comment]
Search box test failures need to be debugged/fixed
```
 3 failing

  1) Search Box Test Suites
       with maxWithoutQuery capping suggestions & recentQueries
         when changing the redirection-url prop, re-initializing the search box
           "before each" hook for "should contain "query-suggestion-1"":
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `[data-query="query-suggestion-1"]`, but never found it. Queried from element: <document>

Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `when changing the redirecti...`
      at Object.querySuggestion (webpack:///./cypress/e2e/search-box/search-box-selectors.ts:15:32)
      at Context.eval (webpack:///./cypress/e2e/search-box/search-box.cypress.ts:150:27)

  2) Search Box Test Suites
       with maxWithoutQuery capping suggestions & recentQueries
         with input
           verify rendering
             should display 10 suggestions:

      AssertionError: Timed out retrying after 4000ms: Not enough elements found. Found '4', expected '10'.
      + expected - actual

      -4
      +10
      
      at Context.eval (webpack:///./cypress/e2e/search-box/search-box-assertions.ts:19:7)

  3) Search Box Test Suites
       with maxWithoutQuery capping suggestions & recentQueries
         with input
           verify rendering
             screen readers should read out "10".:

      Timed out retrying after 4000ms
      + expected - actual

      -'4 search suggestions are available.'
      +'10'
      
      at Context.eval (webpack:///./cypress/e2e/common-assertions.ts:131:15)
```

**Comment by louis-bompart** [comment]
Stale

**Pull request closed without merging**